internal/server/grants_test.go

```diff
@@ -492,30 +492,18 @@ func TestAPI_ListGrants(t *testing.T) {
 				err = json.NewDecoder(resp.Body).Decode(&grants)
 				assert.NilError(t, err)
 
-				expected := []api.Grant{
-					{User: idInGroup, Privilege: "custom1", Resource: "res1"},
-					{User: idOther, Privilege: "custom2", Resource: "res1.ns1"},
-					{User: idOther, Privilege: "connector", Resource: "res1.ns2"},
+				expected := api.ListResponse[api.Grant]{
+					Items: []api.Grant{
+						{User: idInGroup, Privilege: "custom1", Resource: "res1"},
+						{User: idOther, Privilege: "custom2", Resource: "res1.ns1"},
+						{User: idOther, Privilege: "connector", Resource: "res1.ns2"},
+					},
+					Count: 3,
 				}
-				assert.DeepEqual(t, grants.Items, expected, cmpAPIGrantShallow)
+				assert.DeepEqual(t, grants, expected, cmpAPIGrantShallow)
 				assert.Equal(t, resp.Result().Header.Get("Last-Update-Index"), "10004")
 			},
 		},
-		"with latest update index and timeout": {
```

Comment:
I moved this test case to `TestAPI_ListGrants_ExtendedRequestTimeout`. The trick I was using previously to set a context from the in-process client no longer works because of how we extend the request timeout.